app/controllers/sys_controller.rb

```diff
@@ -99,14 +101,37 @@ def check_enabled
   private
 
   def require_basic_auth
+    @authenticated_user =
+    if Setting.repository_authentication_bypass?
```

Comment:
Did you check whether this works as expected?

I believe to remember that the `Setting` class does not work very well for booleans. If that is right, the value could be something like `0` or `1`, which would both be truthy. I direly hope that I am wrong.